src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -129,6 +132,11 @@ public List<FolderIconDescriptor> getIconDescriptors() {
         return r;
     }
 
+    public FormValidation doCheckDisplayNameOrNull(@AncestorInPath AbstractFolder folder, @QueryParameter String value) {
```

Comment:
```suggestion
    @Restricted(NoExternalUse.class)
    public FormValidation doCheckDisplayNameOrNull(@AncestorInPath AbstractFolder folder, @QueryParameter String value) {
```

(Plus imports)

This plugin is a common dependency, and we don't want programmatic callers in other components for stuff like this.

Reply:
Thank you Daniel, applied in [b4821ca](https://github.com/jenkinsci/cloudbees-folder-plugin/pull/268/commits/b4821ca080a49ba8a5874442445040b093386482)